src/model/Space.cpp

```diff
@@ -926,17 +1047,24 @@ namespace model {
 
       auto volumePoly = this->polyhedron();
 
-      auto [isVolEnclosed, edgesNot2] = volumePoly.isEnclosedVolume();
-      if (isVolEnclosed) {
-        return volumePoly.calcPolyhedronVolume();
+      if (volumePoly.isEnclosedVolume()) {
+        if (volumePoly.isCompletelyInsideOut()) {
+          const double volume = volumePoly.polyhedronVolume();
+          LOG(Error, briefDescription() << " has all of its Surfaces that are inside-out. Call Space::fixSurfacesWithIncorrectOrientation().");
+          return -volume;
+        } else if (!volumePoly.hasAnySurfaceWithIncorrectOrientation()) {
+          return volumePoly.polyhedronVolume();
+        } else {
+          LOG(Warn, briefDescription() << " has some Surfaces with incorrection orientation. Call Space::fixSurfacesWithIncorrectOrientation(). "
+                                          "Falling back to ceilingHeight * floorArea. Volume calculation will be potentially inaccurate.");
+        }
+      } else {
+        LOG(Warn, briefDescription() << " is not enclosed, there are " << volumePoly.edgesNotTwo().size()
+                                     << " edges that aren't used exactly twice. "
+                                        "Falling back to ceilingHeight * floorArea. Volume calculation will be potentially inaccurate.");
       }
 
-      LOG(Warn, briefDescription() << " is not enclosed, there are " << edgesNot2.size()
-                                   << " edges that aren't used exactly twice. Volume calculation will be potentially inaccurate");
-
-      double result = 0;
-
-      result = this->ceilingHeight() * this->floorArea();
+      const double result = this->ceilingHeight() * this->floorArea();
```

Comment:
New Space::volume logic to dispatch and LOG whenever the space is non enclosed, or has incorrectly oriented surfaces.